content/blog/2021/05/2021-05-15-configure-plugins-with-jcasc.adoc

```diff
@@ -0,0 +1,115 @@
+---
+:layout: post
+:title: "Configure Plugins with JCasC"
+:tags:
+- jcasc
+- tutorial
+- plugins
+- yaml
+:author: dheerajodha
+description: >
+  This post is a Step-by-Step Guide to configure your plugins with the Jenkins Configuration as a Code (JCasC) plugin  
+opengraph:
+  image: image:/images/post-images/2021-05-15-configure-plugins-with-jcasc/opengraph-image.jpg
+---
+
+This blog post is for anyone interested to know how to configure a plugin using the Jenkins Configuration as a Code (JCasC) plugin, more specifically, this blog will guide you to get the YAML equivalent of a plugin’s configuration and use it to do some changes to the plugin without using the Jenkins UI.
+
+---
+If you’re a beginner at JCasC and want to learn more about it, you can head over to the following resources to understand JCasC better:
+
+* link:https://plugins.jenkins.io/configuration-as-code/[JCasC Documentation]
+* link:https://www.youtube.com/watch?v=PAKWqRE0aTk[Overview of JCasC (Online Meetup)]
+* link:https://www.youtube.com/watch?v=47D3H1BZi4o[Manage JCasC (DevOps World 2018)]
+
+'''
+
+.Configuring your first plugin using JCasC (Video Demo)
+video::YeWhqLPjvMs[youtube, width=852, height=480]
+
+'''
+
+== Overview
+
+So, these are the steps we will be following to achieve our aim:
+
+. Configure the plugin on the UI
+. Download the configuration
+. Update JCasC file locally
+. Load the jenkins.yaml file on the Jenkins server
+. Verify the changes on the UI
```

Comment:
A suggestion to allow asciidoc to generate hyperlinks for the list of contents
```suggestion
. <<Configure the plugin on the UI>>
. <<Download the configuration>>
. <<Update JCasC file locally>>
. <<Load the jenkins.yaml file on the Jenkins server>>
. <<Verify the changes on the UI>>
```

Reply:
This is very interesting, I was looking at how to do this with Asciidoc, thank you!